### ISSUE TITLE
FEATURE: set SVG image sizes by crawling width, height, viewbox attributes from resource stream

### ISSUE THREAD

**Issue by patricekaufmann** [neos/neos-development-collection#3959]
### Is there an existing issue for this topic?

- [X] I have searched the existing issues

### Description

When using svg as images, the ImageService sets width and height to null. This leads to problems when trying to avoid content jumps with loaded svg images. It is neccessary to calculate the image aspect ratio to reserve the required space.

Currently it is not possible because there is no code to actually determine svg dimensions. The ImageService.php in Neos\Media contains 2 `//TODO...`, already 7 years old ;D

It would be nice if you could have a look at the possible solution. Maybe I am missing something important. Otherwise I would elaborate on this and open a PR.

### Possible Solution

This possible solution extends the `getImageSize` function and finalizes the ancient TODO statement. It reads the SVG contents line by line, because most likely the width / height / viewbox parameters would sit at the top:

```
        // TODO: Special handling for SVG should be refactored at a later point.
        if ($resource->getMediaType() === 'image/svg+xml') {

            $imageSize = ['width' => null, 'height' => null];

            $originalResourceStream = $resource->getStream();
            // read single lines as dimension params are most likely to be at the top
            while ($svgLine = fgets($originalResourceStream)) {
                $matches = [];
                if (preg_match('/viewBox=["\']\d* \d* (\d*) (\d*)["\']/', $svgLine, $matches) || preg_match('/width=[\'"](\d)*[\'"].*height=[\'"](\d)*[\'"]/', $svgLine, $matches)) {
                    $imageSize = ['width' => (int)$matches[1], 'height' => (int)$matches[2]];
                    break;
                }
            }
            fclose($originalResourceStream);

        } else {
```

**Comment by Sebobo** [comment]
Good idea to finally fix this. I would parse it as xml as that's what it is. 
There is a comment on stack overflow with a good explanation on the possible variants and also an implementation to calculate the size: https://stackoverflow.com/a/48994280 and https://github.com/contao/imagine-svg/blob/08eb302552dc83ffc7d0f3b1b79f53207094cfa4/src/Image.php#L354

**Comment by mficzel** [comment]
The svg issue is described more broadly in #5652
